# Introducción del archivo

En este archivo, vamos a importar los datos financieros por cada una de las 35 empresas del IBEX 35.

En primer lugar, empezaremos importando gracias a la API de Yahoo Finance los datos de cada una de las firmas. 

En segundo lugar, especificaremos el intervalo temporal que queremos escoger para analizar.

En tercer lugar, vamos a analizar la información resultante del dataset de cada empresa.

Finalmente, guardaremos en formato .csv un dataset por cada una de las 35 compañías.

A lo largo de este archivo, se ha separado en 6 principales pasos los procesos anteriores:
Para la extracción de datos de cada una de las 35 empresas del IBEX 35, vamos a elaborar una serie de pasos:

# Indice de pasos:

- [Paso 1: Instalar las librerias necesarias](#Paso-1:-Instalar-las-librerias-necesarias)
- [Paso 2: Importar las librerias necesarias para nuestro analisis](#Paso-2:-Importar-las-librerias-necesarias-para-nuestro-analisis)
- [Paso 3: Lista de empresas del IBEX 35](#Paso-3:-Lista-de-empresas-del-IBEX-35)
- [Paso 4: Extraccion de datasets de empresas de IBEX 35 a partir de Yahoo Finance](#Paso-4:-Extraccion-de-datasets-de-empresas-de-IBEX-35-a-partir-de-Yahoo-Finance)
    - [Paso 4.1- Obtención de datos financieros de las 35 empresas a partir de Yahoo Finance](#Paso-41--Obtencion-de-datos-financieros-de-las-35-empresas-a-partir-de-Yahoo-Finance)
    - [Paso 4.2- Especificar los parametros de fechas a escoger](#Paso-42--Especificar-los-parametros-de-fechas-a-escoger)
    - [Paso 4.3- Establecer el formato de fecha](#Paso-43--Establecer-el-formato-de-fecha)
- [Paso 5: Informacion acerca del dataset](#Paso-5:-Informacion-acerca-del-dataset)
    - [Paso 5.1- Filas y columnas de cada empresa](#Paso-51--Filas-y-columnas-de-cada-empresa)
    - [Paso 5.2- Variables que vienen en los datos de las empresas a partir de Yahoo Finance](#Paso-52--Variables-que-vienen-en-los-datos-de-las-empresas-a-partir-de-Yahoo-Finance)
    - [Paso 5.3- Informacion de las columnas de los datasets de las empresas](#Paso-53--Informacion-de-las-columnas-de-los-datasets-de-las-empresas)
- [Paso 6: Guardar los datos de Yahoo Finance en archivos CSV. Uno por cada empresa del IBEX-35](#Paso-6:-Guardar-los-datos-de-Yahoo-Finance-en-archivos-CSV.-Uno-por-cada-empresa-del-IBEX-35)


# Paso 1: Instalar las librerias necesarias

In [1]:
#pip install pandas
#pip install yfinance

# Paso 2: Importar las librerias necesarias para nuestro analisis

In [2]:
import yfinance as yf

import pandas as pd
import datetime

# Paso 3: Lista de empresas del IBEX 35

El Ibex 35 es el índice bursátil de referencia de la bolsa española conformado por los 35 valores con más liquidez que cotizan en las cuatro bolsas españolas: Madrid, Barcelona, Bilbao y Valencia. 

A continuación, se van a mencionar por orden alfabético el listado de empresas del IBEX 35.

Esta lista está elaborada con el listado de empresas a fecha de 23/04/2024:

|Número| Compañía | Etiqueta |
|-|----------|----------|
|1| Acciona    | ANA.MC   |
|2| Acciona Energías    | ANE.MC   |
|3| ACS    | ACS.MC  |
|4| Acerinox (NO)    | ACX.MC  |
|5| Aena				|AENA.MC
|6|Amadeus			|AMS.MC
|7|ArcelorMittal		|MTS.MC
|8|Banco Sabadell		|SAB.MC
|9|Banco Santander	|SAN.MC
|10|Bankinter			|BKT.MC
|11|BBVA				|BBVA.MC
|12|CaixaBank			|CABK.MC
|13|Cellnex			|CLNX.MC
|14|Colonial			|COL.MC
|15|Enagás				|ENG.MC
|16|Endesa				|ELE.MC
|17|Ferrovial			|FER.MC
|18|Fluidra			|FDR.MC
|19|Grifols			|GRF.MC
|20|IAG				|IAG.MC
|21|Iberdrola			|IBE.MC
|22|Indra (NO)				|IDR.MC
|23|Inditex			|ITX.MC
|24|Logista 			|LOG.MC
|25|Mapfre				|MAP.MC
|26| Meliá				|MEL.MC
|27|Merlin Properties	|MRL.MC
|28|Naturgy			|NTGY.MC
|29|Redeia				|RED.MC
|30|Repsol				|REP.MC
|31|Rovi				|ROVI.MC
|32|Sacyr				|SCYR.MC
|33|Solaria			|SLR.MC
|34|Telefónica			|TEF.MC
|35|Unicaja			|UNI.MC

# Paso 4: Extraccion de datasets de empresas de IBEX 35 a partir de Yahoo Finance

Vamos a extraer los datos de las empresas del IBEX 35, utilizando la API proporcionada por Yahoo Finance a través de la biblioteca yfinance en Python.

A continuación, vamos a especificar de nuevo la librería a pesar de haberse importado en el Paso 1.

In [3]:
import yfinance as yf

La biblioteca yfinance actúa como un intermediario entre tu código Python y los servidores de Yahoo Finance, facilitando la obtención de datos financieros como precios históricos, datos en tiempo real, información de la empresa, entre otros.

Tras tener importada la librería de Yahoo Finance, cabe la posibilidad de acceder entre otros a los datos financieros de las empresas del IBEX 35.

En principio, tendremos en cuenta los datos en los días operativos entre las fechas de: 01/01/2019 y 30/04/2024. Es decir, 5 años y 4 meses enteros en total.

Hay que tener en cuenta que en la bolsa del IBEX 35, los días dentro de los fines de semana la bolsa está cerrada. En adicción, hay otros tantos días que no son festivos. Por lo que en cada año no habría 365 días.

A continuación, se muestra:
- Paso 4.1: Obtención de datos financieros de las 35 empresas a partir de Yahoo Finance.
- Paso 4.2: Especificar los parámetros de fechas a escoger.
- Paso 4.3: Establecer el formato de fecha.

## Paso 4.1- Obtencion de datos financieros de las 35 empresas a partir de Yahoo Finance

Aquí importamos el Ticker de Yahoo Finance para cada empresa, y le asignamos una variable con su respectivo nombre:

In [4]:
#1 Acciona
ANA_data = yf.Ticker('ANA.MC')
#2 Acciona Energías
ANE_data = yf.Ticker('ANE.MC')
#3 ACS
ACS_data = yf.Ticker('ACS.MC')
#4 Acerinox
ACX_data = yf.Ticker('ACX.MC')
#5 Aena
AENA_data = yf.Ticker('AENA.MC')
#6 Amadeus
AMS_data = yf.Ticker('AMS.MC')
#7 ArcelorMittal
MTS_data = yf.Ticker('MTS.MC')
#8 Banco Sabadell
SAB_data = yf.Ticker('SAB.MC')
#9 Banco Santander
SAN_data = yf.Ticker('SAN.MC')
#10 Bankinter
BKT_data = yf.Ticker('BKT.MC')
#11 BBVA
BBVA_data = yf.Ticker('BBVA.MC')
#12 CaixaBank
CABK_data = yf.Ticker('CABK.MC')
#13 Cellnex
CLNX_data = yf.Ticker('CLNX.MC')
#14 Colonial
COL_data = yf.Ticker('COL.MC')
#15 Enagás
ENG_data = yf.Ticker('ENG.MC')
#16 Endesa	
ELE_data = yf.Ticker('ELE.MC')
#17 Ferrovial
FER_data = yf.Ticker('FER.MC')
#18 Fluidra
FDR_data = yf.Ticker('FDR.MC')
#19 Grifols
GRF_data = yf.Ticker('GRF.MC')
#20 IAG	
IAG_data = yf.Ticker('IAG.MC')
#21 Iberdrola
IBE_data = yf.Ticker('IBE.MC')
#22 Indra
IDR_data = yf.Ticker('IDR.MC')
#23 Inditex
ITX_data = yf.Ticker('ITX.MC')
#24 Logista
LOG_data = yf.Ticker('LOG.MC')
#25 Mapfre
MAP_data = yf.Ticker('MAP.MC')
#26 Meliá
MEL_data = yf.Ticker('MEL.MC')
#27 Merlin Properties
MRL_data = yf.Ticker('MRL.MC')
#28 Naturgy
NTGY_data = yf.Ticker('NTGY.MC')
#29 Redeia
RED_data = yf.Ticker('RED.MC')
#30 Repsol
REP_data = yf.Ticker('REP.MC')
#31 Rovi
ROVI_data = yf.Ticker('ROVI.MC')
#32 Sacyr
SCYR_data = yf.Ticker('SCYR.MC')
#33 Solaria
SLR_data = yf.Ticker('SLR.MC')
#34 Telefónica
TEF_data = yf.Ticker('TEF.MC')
#35 Unicaja
UNI_data = yf.Ticker('UNI.MC')

## Paso 4.2- Especificar los parametros de fechas a escoger

Vamos a tener en cuenta, que el intervalo temporal a tener en cuenta para la elaboración de los análisis, tendrá por defecto las siguientes fechas:
- La Fecha inicial se establece en el 01/01/2019
- La Fecha final se establece en el 30/04/2024

En cualquier caso, estas fechas se pueden modificar en la siguiente celda a gusto del usuario.

In [5]:
import datetime

#1   fecha_inicial, la cual podemos modificar a nuestro antojo
fecha_inicial = '2019-01-01'

#2   fecha_final, la cual podemos modificar a nuestro antojo
fecha_final = '2024-04-30'

Por otra parte, dejamos especificados los parámetros de start y end, con la fecha inicial y final establecida en la celda anterior.

A continuación, mostramos como quedaría el filtrado de las fechas mediante la aplicación para la empresa ACS:

In [6]:
ACS_df = ACS_data.history(start=fecha_inicial, end=fecha_final)
ACS_df

Open       High        Low      Close  \
Date                                                                    
2019-01-02 00:00:00+01:00  22.949236  22.949236  22.233562  22.656149   
2019-01-03 00:00:00+01:00  22.424413  22.921974  22.356254  22.710682   
2019-01-04 00:00:00+01:00  22.969682  23.167346  22.881075  23.051474   
2019-01-07 00:00:00+01:00  23.242324  23.378642  22.921974  23.126451   
2019-01-08 00:00:00+01:00  23.119633  23.603565  23.065107  23.365009   
...                              ...        ...        ...        ...   
2024-04-23 00:00:00+02:00  38.500000  38.900002  38.459999  38.480000   
2024-04-24 00:00:00+02:00  38.520000  38.700001  38.220001  38.360001   
2024-04-25 00:00:00+02:00  38.480000  38.480000  37.360001  37.700001   
2024-04-26 00:00:00+02:00  38.040001  38.080002  37.700001  37.820000   
2024-04-29 00:00:00+02:00  37.799999  38.139999  37.799999  37.980000   

                            Volume  Dividends  Stock Splits  
Date                                                         
2019-01-02 00:00:00+01:00   579957        0.0           0.0  
2019-01-03 00:00:00+01:00   826120        0.0           0.0  
2019-01-04 00:00:00+01:00   962645        0.0           0.0  
2019-01-07 00:00:00+01:00  1501743        0.0           0.0  
2019-01-08 00:00:00+01:00  1264760        0.0           0.0  
...                            ...        ...           ...  
2024-04-23 00:00:00+02:00   391787        0.0           0.0  
2024-04-24 00:00:00+02:00   732447        0.0           0.0  
2024-04-25 00:00:00+02:00   503456        0.0           0.0  
2024-04-26 00:00:00+02:00   522897        0.0           0.0  
2024-04-29 00:00:00+02:00   341569        0.0           0.0  

[1364 rows x 7 columns]

En adicción, se muestra el mismo procedimiento para cada empresa del IBEX 35:

In [7]:
ANA_df = ANA_data.history(start=fecha_inicial, end=fecha_final)
ANE_df = ANE_data.history(start=fecha_inicial, end=fecha_final)
ACS_df = ACS_data.history(start=fecha_inicial, end=fecha_final)
ACX_df = ACX_data.history(start=fecha_inicial, end=fecha_final)
AENA_df = AENA_data.history(start=fecha_inicial, end=fecha_final)
AMS_df = AMS_data.history(start=fecha_inicial, end=fecha_final)
MTS_df = MTS_data.history(start=fecha_inicial, end=fecha_final)
SAB_df = SAB_data.history(start=fecha_inicial, end=fecha_final)
SAN_df = SAN_data.history(start=fecha_inicial, end=fecha_final)
BKT_df = BKT_data.history(start=fecha_inicial, end=fecha_final)
BBVA_df = BBVA_data.history(start=fecha_inicial, end=fecha_final)
CABK_df = CABK_data.history(start=fecha_inicial, end=fecha_final)
CLNX_df = CLNX_data.history(start=fecha_inicial, end=fecha_final)
COL_df = COL_data.history(start=fecha_inicial, end=fecha_final)
ENG_df = ENG_data.history(start=fecha_inicial, end=fecha_final)
ELE_df = ELE_data.history(start=fecha_inicial, end=fecha_final)
FER_df = FER_data.history(start=fecha_inicial, end=fecha_final)
FDR_df = FDR_data.history(start=fecha_inicial, end=fecha_final)
GRF_df = GRF_data.history(start=fecha_inicial, end=fecha_final)
IAG_df = IAG_data.history(start=fecha_inicial, end=fecha_final)
IBE_df = IBE_data.history(start=fecha_inicial, end=fecha_final)
IDR_df = IDR_data.history(start=fecha_inicial, end=fecha_final)
ITX_df = ITX_data.history(start=fecha_inicial, end=fecha_final)
LOG_df = LOG_data.history(start=fecha_inicial, end=fecha_final)
MAP_df = MAP_data.history(start=fecha_inicial, end=fecha_final)
MEL_df = MEL_data.history(start=fecha_inicial, end=fecha_final)
MRL_df = MRL_data.history(start=fecha_inicial, end=fecha_final)
NTGY_df = NTGY_data.history(start=fecha_inicial, end=fecha_final)
RED_df = RED_data.history(start=fecha_inicial, end=fecha_final)
REP_df = REP_data.history(start=fecha_inicial, end=fecha_final)
ROVI_df = ROVI_data.history(start=fecha_inicial, end=fecha_final)
SCYR_df = SCYR_data.history(start=fecha_inicial, end=fecha_final)
SLR_df = SLR_data.history(start=fecha_inicial, end=fecha_final)
TEF_df = TEF_data.history(start=fecha_inicial, end=fecha_final)
UNI_df = UNI_data.history(start=fecha_inicial, end=fecha_final)

## Paso 4.3- Establecer el formato de fecha

In [8]:
ACS_df

Open       High        Low      Close  \
Date                                                                    
2019-01-02 00:00:00+01:00  22.949236  22.949236  22.233562  22.656149   
2019-01-03 00:00:00+01:00  22.424413  22.921974  22.356254  22.710682   
2019-01-04 00:00:00+01:00  22.969682  23.167346  22.881075  23.051474   
2019-01-07 00:00:00+01:00  23.242324  23.378642  22.921974  23.126451   
2019-01-08 00:00:00+01:00  23.119633  23.603565  23.065107  23.365009   
...                              ...        ...        ...        ...   
2024-04-23 00:00:00+02:00  38.500000  38.900002  38.459999  38.480000   
2024-04-24 00:00:00+02:00  38.520000  38.700001  38.220001  38.360001   
2024-04-25 00:00:00+02:00  38.480000  38.480000  37.360001  37.700001   
2024-04-26 00:00:00+02:00  38.040001  38.080002  37.700001  37.820000   
2024-04-29 00:00:00+02:00  37.799999  38.139999  37.799999  37.980000   

                            Volume  Dividends  Stock Splits  
Date                                                         
2019-01-02 00:00:00+01:00   579957        0.0           0.0  
2019-01-03 00:00:00+01:00   826120        0.0           0.0  
2019-01-04 00:00:00+01:00   962645        0.0           0.0  
2019-01-07 00:00:00+01:00  1501743        0.0           0.0  
2019-01-08 00:00:00+01:00  1264760        0.0           0.0  
...                            ...        ...           ...  
2024-04-23 00:00:00+02:00   391787        0.0           0.0  
2024-04-24 00:00:00+02:00   732447        0.0           0.0  
2024-04-25 00:00:00+02:00   503456        0.0           0.0  
2024-04-26 00:00:00+02:00   522897        0.0           0.0  
2024-04-29 00:00:00+02:00   341569        0.0           0.0  

[1364 rows x 7 columns]

A continuación, vamos a establecer el formato de fecha %Y-%m-%d , que es básicamente el YYYY-MM-DD. 

Este formato no está asociado específicamente a un país en particular, sino que es un estándar de formato de fecha ampliamente utilizado en la programación y en sistemas informáticos en general. Es parte de la especificación de formateo de fecha y hora de la biblioteca estándar de Python y se utiliza en muchas otras herramientas y lenguajes de programación.

A continuación, lo aplicamos a los dataframes de las 35 empresas:

In [9]:
ANA_df.index = ANA_df.index.strftime('%Y-%m-%d')
ANE_df.index = ANE_df.index.strftime('%Y-%m-%d')
ACS_df.index = ACS_df.index.strftime('%Y-%m-%d')
ACX_df.index = ACX_df.index.strftime('%Y-%m-%d')
AENA_df.index = AENA_df.index.strftime('%Y-%m-%d')
AMS_df.index = AMS_df.index.strftime('%Y-%m-%d')
MTS_df.index = MTS_df.index.strftime('%Y-%m-%d')
SAB_df.index = SAB_df.index.strftime('%Y-%m-%d')
SAN_df.index = SAN_df.index.strftime('%Y-%m-%d')
BKT_df.index = BKT_df.index.strftime('%Y-%m-%d')
BBVA_df.index = BBVA_df.index.strftime('%Y-%m-%d')
CABK_df.index = CABK_df.index.strftime('%Y-%m-%d')
CLNX_df.index = CLNX_df.index.strftime('%Y-%m-%d')
COL_df.index = COL_df.index.strftime('%Y-%m-%d')
ENG_df.index = ENG_df.index.strftime('%Y-%m-%d')
ELE_df.index = ELE_df.index.strftime('%Y-%m-%d')
FER_df.index = FER_df.index.strftime('%Y-%m-%d')
FDR_df.index = FDR_df.index.strftime('%Y-%m-%d')
GRF_df.index = GRF_df.index.strftime('%Y-%m-%d')
IAG_df.index = IAG_df.index.strftime('%Y-%m-%d')
IBE_df.index = IBE_df.index.strftime('%Y-%m-%d')
IDR_df.index = IDR_df.index.strftime('%Y-%m-%d')
ITX_df.index = ITX_df.index.strftime('%Y-%m-%d')
LOG_df.index = LOG_df.index.strftime('%Y-%m-%d')
MAP_df.index = MAP_df.index.strftime('%Y-%m-%d')
MEL_df.index = MEL_df.index.strftime('%Y-%m-%d')
MRL_df.index = MRL_df.index.strftime('%Y-%m-%d')
NTGY_df.index = NTGY_df.index.strftime('%Y-%m-%d')
RED_df.index = RED_df.index.strftime('%Y-%m-%d')
REP_df.index = REP_df.index.strftime('%Y-%m-%d')
ROVI_df.index = ROVI_df.index.strftime('%Y-%m-%d')
SCYR_df.index = SCYR_df.index.strftime('%Y-%m-%d')
SLR_df.index = SLR_df.index.strftime('%Y-%m-%d')
TEF_df.index = TEF_df.index.strftime('%Y-%m-%d')
UNI_df.index = UNI_df.index.strftime('%Y-%m-%d')

De forma que quede cada empresa de la siguiente forma:

In [10]:
ACS_df

Open       High        Low      Close   Volume  Dividends  \
Date                                                                         
2019-01-02  22.949236  22.949236  22.233562  22.656149   579957        0.0   
2019-01-03  22.424413  22.921974  22.356254  22.710682   826120        0.0   
2019-01-04  22.969682  23.167346  22.881075  23.051474   962645        0.0   
2019-01-07  23.242324  23.378642  22.921974  23.126451  1501743        0.0   
2019-01-08  23.119633  23.603565  23.065107  23.365009  1264760        0.0   
...               ...        ...        ...        ...      ...        ...   
2024-04-23  38.500000  38.900002  38.459999  38.480000   391787        0.0   
2024-04-24  38.520000  38.700001  38.220001  38.360001   732447        0.0   
2024-04-25  38.480000  38.480000  37.360001  37.700001   503456        0.0   
2024-04-26  38.040001  38.080002  37.700001  37.820000   522897        0.0   
2024-04-29  37.799999  38.139999  37.799999  37.980000   341569        0.0   

            Stock Splits  
Date                      
2019-01-02           0.0  
2019-01-03           0.0  
2019-01-04           0.0  
2019-01-07           0.0  
2019-01-08           0.0  
...                  ...  
2024-04-23           0.0  
2024-04-24           0.0  
2024-04-25           0.0  
2024-04-26           0.0  
2024-04-29           0.0  

[1364 rows x 7 columns]

# Paso 5: Informacion acerca del dataset
Vamos a analizar la información que nos viene de las 35 compañías.

## Paso 5.1- Filas y columnas de cada empresa
En primer, lugar tenemos las siguientes filas y columnas para ACS como ejemplo:

In [11]:
ACS_df.shape

(1364, 7)

Tenemos un total de:
- 1364 filas
- el index de 'Date' y 7 columnas

Con el output de la siguiente celda, vamos a ver las filas y columnas de todas las compañías del IBEX-35:

In [12]:
dataframes_dict = {
    "ANA_df": ANA_df,
    "ANE_df": ANE_df,
    "ACS_df": ACS_df,
    "ACX_df": ACX_df,
    "AENA_df": AENA_df,
    "AMS_df": AMS_df,
    "MTS_df": MTS_df,
    "SAB_df": SAB_df,
    "SAN_df": SAN_df,
    "BKT_df": BKT_df,
    "BBVA_df": BBVA_df,
    "CABK_df": CABK_df,
    "CLNX_df": CLNX_df,
    "COL_df": COL_df,
    "ENG_df": ENG_df,
    "ELE_df": ELE_df,
    "FER_df": FER_df,
    "FDR_df": FDR_df,
    "GRF_df": GRF_df,
    "IAG_df": IAG_df,
    "IBE_df": IBE_df,
    "IDR_df": IDR_df,
    "ITX_df": ITX_df,
    "LOG_df": LOG_df,
    "MAP_df": MAP_df,
    "MEL_df": MEL_df,
    "MRL_df": MRL_df,
    "NTGY_df": NTGY_df,
    "RED_df": RED_df,
    "REP_df": REP_df,
    "ROVI_df": ROVI_df,
    "SCYR_df": SCYR_df,
    "SLR_df": SLR_df,
    "TEF_df": TEF_df,
    "UNI_df": UNI_df
}

for i, (df_name, df) in enumerate(dataframes_dict.items(), start=1):
    print(f"{i}. {df_name}: {df.shape}")

1. ANA_df: (1364, 7)
2. ANE_df: (724, 7)
3. ACS_df: (1364, 7)
4. ACX_df: (1364, 7)
5. AENA_df: (1364, 7)
6. AMS_df: (1364, 7)
7. MTS_df: (1364, 7)
8. SAB_df: (1364, 7)
9. SAN_df: (1364, 7)
10. BKT_df: (1364, 7)
11. BBVA_df: (1364, 7)
12. CABK_df: (1364, 7)
13. CLNX_df: (1364, 7)
14. COL_df: (1364, 7)
15. ENG_df: (1364, 7)
16. ELE_df: (1364, 7)
17. FER_df: (1364, 7)
18. FDR_df: (1364, 7)
19. GRF_df: (1364, 7)
20. IAG_df: (1364, 7)
21. IBE_df: (1364, 7)
22. IDR_df: (1364, 7)
23. ITX_df: (1364, 7)
24. LOG_df: (1364, 7)
25. MAP_df: (1364, 7)
26. MEL_df: (1364, 7)
27. MRL_df: (1364, 7)
28. NTGY_df: (1364, 7)
29. RED_df: (1364, 7)
30. REP_df: (1364, 7)
31. ROVI_df: (1364, 7)
32. SCYR_df: (1364, 7)
33. SLR_df: (1364, 7)
34. TEF_df: (1364, 7)
35. UNI_df: (1364, 7)


De entre las 35 compañías, se ha identificado una empresa con menos filas:

In [13]:
ANE_df.shape    # Acciona Energia tiene menor cantidad de filas.
#Salió a bolsa en agosto del 2021

(724, 7)

## Paso 5.2- Variables que vienen en los datos de las empresas a partir de Yahoo Finance

En segundo lugar, desde Yahoo Finance, estamos importando de cada empresa las siguientes variables:

Como podemos observar, hay 7 columnas que vamos a explicar brevemente.

1- Date (index)

Esta columna la hemos creado previamente. Date(fecha) se refiere al punto específico en el tiempo en el que se lleva a cabo una transacción o una serie de transacciones en el mercado de valores. La fecha se indica generalmente en un formato estándar de día, mes y año y es crucial para el registro preciso de las operaciones financieras.

2- Open

Open (Apertura) describe el precio al que una acción comienza a ser negociada al inicio de un periodo determinado de cotización, como el inicio de una jornada de trading o el inicio de un intervalo de tiempo específico.

3- High

High (Alto) se refiere al precio más alto alcanzado por una acción durante el periodo de tiempo analizado. Representa el punto máximo de valoración alcanzado por la acción en el mercado durante ese periodo.

4- Low

Low (Baja) es el precio más bajo registrado para una acción durante el periodo de tiempo en cuestión. Indica el nivel mínimo de valoración al que la acción ha llegado en el mercado durante ese periodo de tiempo.

5- Close

Close (Cierre) describe el precio al que una acción finaliza su cotización al final de un periodo específico, como el cierre de una jornada de trading o el final de un intervalo de tiempo determinado.

6- Volume

Volume (Volumen) indica el número total de acciones que se han negociado durante un periodo de tiempo específico. El volumen es un indicador clave de la actividad de trading en el mercado de valores y puede proporcionar información sobre el interés y la liquidez de una acción en particular.

7- Dividends

Dividends (Dividendos) indica los dividendos pagados por las acciones de la empresa en la fecha correspondiente, si los hubiera. Los dividendos representan los pagos periódicos realizados a los accionistas como una distribución de las ganancias de la empresa.

8- Stock Splits
 
Stock Splits (División de acciones) indica si se produjeron divisiones de acciones de ACS en la fecha indicada. Un split de acciones implica la división de las acciones existentes en múltiples acciones nuevas, generalmente con el objetivo de ajustar el precio por acción y hacerlas más accesibles para los inversores.

## Paso 5.3- Informacion de las columnas de los datasets de las empresas

En tercer lugar, vamos a obtener más información de las columnas:

In [14]:
ACS_df.columns

Index(['Open', 'High', 'Low', 'Close', 'Volume', 'Dividends', 'Stock Splits'], dtype='object')

In [15]:
ACS_df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 1364 entries, 2019-01-02 to 2024-04-29
Data columns (total 7 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Open          1364 non-null   float64
 1   High          1364 non-null   float64
 2   Low           1364 non-null   float64
 3   Close         1364 non-null   float64
 4   Volume        1364 non-null   int64  
 5   Dividends     1364 non-null   float64
 6   Stock Splits  1364 non-null   float64
dtypes: float64(6), int64(1)
memory usage: 85.2+ KB


Comprobar si hay valores NaN en las columnas de los datasets:

In [16]:
ACS_df.isna().sum()

Open            0
High            0
Low             0
Close           0
Volume          0
Dividends       0
Stock Splits    0
dtype: int64

Y a continuación, si los hay a lo largo de todo el dataset...

In [17]:
import pandas as pd

# Diccionario para mapear el nombre del DataFrame con su respectivo DataFrame
dataframes_dict = {
    "ANA_df": ANA_df,
    "ANE_df": ANE_df,
    "ACS_df": ACS_df,
    "ACX_df": ACX_df,
    "AENA_df": AENA_df,
    "AMS_df": AMS_df,
    "MTS_df": MTS_df,
    "SAB_df": SAB_df,
    "SAN_df": SAN_df,
    "BKT_df": BKT_df,
    "BBVA_df": BBVA_df,
    "CABK_df": CABK_df,
    "CLNX_df": CLNX_df,
    "COL_df": COL_df,
    "ENG_df": ENG_df,
    "ELE_df": ELE_df,
    "FER_df": FER_df,
    "FDR_df": FDR_df,
    "GRF_df": GRF_df,
    "IAG_df": IAG_df,
    "IBE_df": IBE_df,
    "IDR_df": IDR_df,
    "ITX_df": ITX_df,
    "LOG_df": LOG_df,
    "MAP_df": MAP_df,
    "MEL_df": MEL_df,
    "MRL_df": MRL_df,
    "NTGY_df": NTGY_df,
    "RED_df": RED_df,
    "REP_df": REP_df,
    "ROVI_df": ROVI_df,
    "SCYR_df": SCYR_df,
    "SLR_df": SLR_df,
    "TEF_df": TEF_df,
    "UNI_df": UNI_df
}

na_counts_dict = {}

for df_name, df in dataframes_dict.items():
    na_counts_dict[df_name] = df.isna().sum()

na_counts_df = pd.DataFrame(na_counts_dict)

# Transponer el df para tener los DataFrames en columnas verticales
na_counts_df = na_counts_df.transpose()

print(na_counts_df)

         Open  High  Low  Close  Volume  Dividends  Stock Splits
ANA_df      0     0    0      0       0          0             0
ANE_df      0     0    0      0       0          0             0
ACS_df      0     0    0      0       0          0             0
ACX_df      0     0    0      0       0          0             0
AENA_df     0     0    0      0       0          0             0
AMS_df      0     0    0      0       0          0             0
MTS_df      0     0    0      0       0          0             0
SAB_df      0     0    0      0       0          0             0
SAN_df      0     0    0      0       0          0             0
BKT_df      0     0    0      0       0          0             0
BBVA_df     0     0    0      0       0          0             0
CABK_df     0     0    0      0       0          0             0
CLNX_df     0     0    0      0       0          0             0
COL_df      0     0    0      0       0          0             0
ENG_df      0     0    0 

Como podemos observar, no hay valores nulos, por lo que no será necesaria una limpieza de los datasets.

Podemos confirmar que los datos que provienen de Yahoo Finance vienen de forma relativamente limpia para poder analizarlos posteriormente.

# Paso 6: Guardar los datos de Yahoo Finance en archivos CSV. Uno por cada empresa del IBEX 35

Tras haber filtrado los 35 datasets, vamos a proceder a guardar cada dataset en archivos .csv

In [18]:
#ANA_df.to_csv('../Data/ANA_dataset.csv')
#ANE_df.to_csv('../Data/ANE_dataset.csv')
#ACS_df.to_csv('../Data/ACS_dataset.csv')
#ACX_df.to_csv('../Data/ACX_dataset.csv')
#AENA_df.to_csv('../Data/AENA_dataset.csv')
#AMS_df.to_csv('../Data/AMS_dataset.csv')
#MTS_df.to_csv('../Data/MTS_dataset.csv')
#SAB_df.to_csv('../Data/SAB_dataset.csv')
#SAN_df.to_csv('../Data/SAN_dataset.csv')
#BKT_df.to_csv('../Data/BKT_dataset.csv')
#BBVA_df.to_csv('../Data/BBVA_dataset.csv')
#CABK_df.to_csv('../Data/CABK_dataset.csv')
#CLNX_df.to_csv('../Data/CLNX_dataset.csv')
#COL_df.to_csv('../Data/COL_dataset.csv')
#ENG_df.to_csv('../Data/ENG_dataset.csv')
#ELE_df.to_csv('../Data/ELE_dataset.csv')
#FER_df.to_csv('../Data/FER_dataset.csv')
#FDR_df.to_csv('../Data/FDR_dataset.csv')
#GRF_df.to_csv('../Data/GRF_dataset.csv')
#IAG_df.to_csv('../Data/IAG_dataset.csv')
#IBE_df.to_csv('../Data/IBE_dataset.csv')
#IDR_df.to_csv('../Data/IDR_dataset.csv')
#ITX_df.to_csv('../Data/ITX_dataset.csv')
#LOG_df.to_csv('../Data/LOG_dataset.csv')
#MAP_df.to_csv('../Data/MAP_dataset.csv')
#MEL_df.to_csv('../Data/MEL_dataset.csv')
#MRL_df.to_csv('../Data/MRL_dataset.csv')
#NTGY_df.to_csv('../Data/NTGY_dataset.csv')
#RED_df.to_csv('../Data/RED_dataset.csv')
#REP_df.to_csv('../Data/REP_dataset.csv')
#ROVI_df.to_csv('../Data/ROVI_dataset.csv')
#SCYR_df.to_csv('../Data/SCYR_dataset.csv')
#SLR_df.to_csv('../Data/SLR_dataset.csv')
#TEF_df.to_csv('../Data/TEF_dataset.csv')
#UNI_df.to_csv('../Data/UNI_dataset.csv')

Aquí ya tenemos los 35 archivos guardados en su correspondiente repositorio.

In [19]:
ACS_df

Open       High        Low      Close   Volume  Dividends  \
Date                                                                         
2019-01-02  22.949236  22.949236  22.233562  22.656149   579957        0.0   
2019-01-03  22.424413  22.921974  22.356254  22.710682   826120        0.0   
2019-01-04  22.969682  23.167346  22.881075  23.051474   962645        0.0   
2019-01-07  23.242324  23.378642  22.921974  23.126451  1501743        0.0   
2019-01-08  23.119633  23.603565  23.065107  23.365009  1264760        0.0   
...               ...        ...        ...        ...      ...        ...   
2024-04-23  38.500000  38.900002  38.459999  38.480000   391787        0.0   
2024-04-24  38.520000  38.700001  38.220001  38.360001   732447        0.0   
2024-04-25  38.480000  38.480000  37.360001  37.700001   503456        0.0   
2024-04-26  38.040001  38.080002  37.700001  37.820000   522897        0.0   
2024-04-29  37.799999  38.139999  37.799999  37.980000   341569        0.0   

            Stock Splits  
Date                      
2019-01-02           0.0  
2019-01-03           0.0  
2019-01-04           0.0  
2019-01-07           0.0  
2019-01-08           0.0  
...                  ...  
2024-04-23           0.0  
2024-04-24           0.0  
2024-04-25           0.0  
2024-04-26           0.0  
2024-04-29           0.0  

[1364 rows x 7 columns]